src/servers/http_server.cc

```diff
@@ -69,6 +71,49 @@ class HTTPAPIServer : public HTTPServer {
       const std::vector<std::string>& endpoints, const int32_t port);
   ~HTTPAPIServer();
 
+  using ResponseMetaData = std::pair<
```

Comment:
Use a proper class for this. See grpc_server.cc for how the response meta-data is handled and use an equivalent class/struct here. In general, parallel as much as possible what is in grpc_server.cc.